nixos/modules/config/networking.nix

```diff
@@ -20,12 +20,35 @@ in
 
   options = {
 
+    networking.fqdn = lib.mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "foo.example.com";
+      description = ''
+        Fully qualified domain name, if any.
+      '';
+    };
+
+    networking.hosts = lib.mkOption {
+      type = types.attrsOf ( types.listOf types.str );
+      default = {};
+      example = literalExample ''
+        {
+          "127.0.0.1" = [ "foo.bar.baz" ];
+          "192.168.0.2" = [ "fileserver.local" "nameserver.local" ];
+        };
+      '';
+      description = ''
+        Locally defined maps of hostnames to IP addresses.
+      '';
+    };
+
     networking.extraHosts = lib.mkOption {
       type = types.lines;
       default = "";
       example = "192.168.0.1 lanlocalhost";
```

Comment:
It might have another example, because ```192.168.0.1 lanlocalhost``` is what is better to set via ```networking.hosts```
In case is there is no other meaningful example for  ```networking.extraHosts``` (except of plain forms of what is to be set in ```networking.hosts```), it might have sense to remove or deprecate this option.